ezomero/ezomero.py

```diff
@@ -66,17 +66,28 @@ def post_dataset(conn, dataset_name, project_id=None, description=None):
     if type(description) is not str and description is not None:
```

Comment:
Apologies for being kinda picky on this PR, but I think this will largely be the template of the changes we'll end up doing to a bunch of functions and I want this to be all the way there before merging!

Reply:
I can probably get to this and make tweaks a bit later today